modules/tinymce/src/core/test/ts/browser/util/I18nTest.ts

```diff
@@ -88,4 +88,37 @@ describe('browser.tinymce.core.util.I18nTest', () => {
 
     I18n.setCode('en');
   });
+
+  context('Case sensitive translations', () => {
+    const addTranslationsAndSetCode = (code: string, translations: Record<string, string>) => {
+      I18n.add(code, translations);
+      I18n.setCode(code);
+    };
+
+    const assertTranslation = (key: string, expectedTranslation: string) =>
+      assert.equal(I18n.translate(key), expectedTranslation);
+
+    it('All case sensitive variants of duplicate keys should be preserved where a lowercase variant is already present', () => {
+      addTranslationsAndSetCode('code-ci1', {
+        test: 'test',
+        Test: 'Test',
+        TEST: 'TEST'
+      });
+
+      assertTranslation('test', 'test');
+      assertTranslation('Test', 'Test');
+      assertTranslation('TEST', 'TEST');
+    });
+
+    it('All case sensitive variants of duplicate keys should be preserved and a lowercase variant should be added where a lowercase variant is not present', () => {
+      addTranslationsAndSetCode('code-ci1', {
+        Test: 'Test',
+        TEST: 'TEST'
+      });
+
+      assert.oneOf(I18n.translate('test'), [ 'Test', 'TEST' ]);
```

Comment:
Can we be more specific than `assert.oneOf`? The translation for the lowercase key is expected to be the last entry i.e. 'TEST' wouldn't it?

Reply:
In the case where a lowercase variant of duplicated keys is not specified, we would create a lowercase variant using the value of whichever variant of the duplicate keys that come first depending on how the browser ordered the object. Since object order is not guaranteed to be retained, it can be either of those two values.